### ISSUE TITLE
How to retrain a image-condition model on my own dataset?

### ISSUE THREAD

**Issue by pokameng** [huggingface/diffusers#1609]
@kig @kashif @srush @StoneCypher @keturn 
Hello everyone!!!
I have facused on this repo long time and had difficulted in retraining a image-condition model on  my own dataset.
I want to etrain a image-condition model on my own dataset ,but i can not find a config suit for my task.
Maybe you can share a example config or guidance link for me?\
Thanks!!!

**Comment by patrickvonplaten** [comment]
Hey @pokameng,

We would need a reproducible code snippet or more context to answer your question here . Thanks :-)

**Comment by pokameng** [comment]
> Hey @pokameng,
> 
> We would need a reproducible code snippet or more context to answer your question here . Thanks :-)

hello，thanks for you reply

This is my config.yaml:
`model:
  base_learning_rate: 1.0e-04
  target: ldm.models.diffusion.ddpm.LatentDiffusion
  params:
    linear_start: 0.00085
    linear_end: 0.0120
    num_timesteps_cond: 1
    log_every_t: 200
    timesteps: 1000
    first_stage_key: image
    cond_stage_key: image
    image_size: 64
    channels: 4
    cond_stage_trainable: false   # Note: different from the one we trained before
    conditioning_key: crossattn
    monitor: val/loss_simple_ema
    scale_factor: 0.18215

    scheduler_config: # 10000 warmup steps
      target: ldm.lr_scheduler.LambdaLinearScheduler
      params:
        warm_up_steps: [ 1000 ]
        cycle_lengths: [ 10000000000000 ] # incredibly large number to prevent corner cases
        f_start: [ 1.e-6 ]
        f_max: [ 1. ]
        f_min: [ 1. ]

    unet_config:
      target: ldm.modules.diffusionmodules.openaimodel.UNetModel
      params:
        image_size: 32 # unused
        in_channels: 4
        out_channels: 4
        model_channels: 320
        attention_resolutions: [ 4, 2, 1 ]
        num_res_blocks: 2
        channel_mult: [ 1, 2, 4, 4 ]
        num_heads: 8
        use_spatial_transformer: True
        transformer_depth: 1
        context_dim: 768
        use_checkpoint: True
        legacy: False

    first_stage_config:
      target:  ldm.models.autoencoder.AutoencoderKL #ldm.models.autoencoder.VQModelInterface    #ldm.models.autoencoder.AutoencoderKL
      params:
        embed_dim: 4
        monitor: val/rec_loss
        ddconfig:
          double_z: true
          z_channels: 4
          resolution: 256
          in_channels: 3
          out_ch: 3
          ch: 128
          ch_mult:
          - 1
          - 2
          - 4
          - 4
          num_res_blocks: 2
          attn_resolutions: []
          dropout: 0.0
        lossconfig:
          target: torch.nn.Identity

    cond_stage_config:
      target: ldm.modules.encoders.modules.FrozenCLIPImageEmbedder


data:
  target: main.DataModuleFromConfig
  params:
    batch_size: 2
    num_workers: 2
    num_val_workers: 0 # Avoid a weird val dataloader issue
    train:
      target: ldm.data.simple.FolderData
      params:
        root_dir: /home/share/movie_dataset/fanghua/png
        ext: jpg
        image_transforms:
        - target: torchvision.transforms.Resize
          params:
            size: 256
            interpolation: 3
        - target: torchvision.transforms.RandomHorizontalFlip
    validation:
      target: ldm.data.simple.FolderData
      params:
        root_dir: /home/share/movie_dataset/fanghua/png
        ext: jpg
        image_transforms:
        - target: torchvision.transforms.Resize
          params:
            size: 256
            interpolation: 3

lightning:
  find_unused_parameters: false
  modelcheckpoint:
    params:
      every_n_train_steps: 5000
  callbacks:
    image_logger:
      target: main.ImageLogger
      params:
        batch_frequency: 1000
        max_images: 8
        increase_log_steps: False
        log_first_step: True
        log_images_kwargs:
          use_ema_scope: False
          inpaint: False
          plot_progressive_rows: False
          plot_diffusion_rows: False
          N: 8
          unconditional_guidance_scale: 3.0
          unconditional_guidance_label: [""]

  trainer:
    benchmark: True
    # val_check_interval: 5000000 # really sorry
    num_sanity_val_steps: 0
    accumulate_grad_batches: 1
`

When I use this  I get condition image :
![image](https://user-images.githubusercontent.com/47957015/207086252-4375c8c4-29ab-4620-80a4-1c6655c0af7c.png)
and input image:
![image](https://user-images.githubusercontent.com/47957015/207086366-a44538a4-58c3-47e8-b450-1e574652e319.png)
but the` reconstruction_gs` is
![image](https://user-images.githubusercontent.com/47957015/207086499-0947d1bd-0ac8-4bdb-a9ca-c59dcd945768.png)
the `sample` is :
![image](https://user-images.githubusercontent.com/47957015/207086594-86a7e463-6f53-4cb5-a717-20f7a0c10f3b.png)

And I want to know, where is the condition image feeded into and where is the input feeded into?
I want to set RGB image as my image condition and gray image as inputs, so how can i set the config.yaml?
My dataset is fanghua from a movie.

**Comment by pokameng** [comment]
@patrickvonplaten

**Comment by patrickvonplaten** [comment]
Hey @pokameng,

Note that: https://github.com/huggingface/diffusers/issues/1609#issuecomment-1346730811 looks like you are using the original CompVis repo code. In this case could you please file an issue there: https://github.com/CompVis/stable-diffusion

**Comment by GISStd** [comment]
I want to do similiar work too.And I want to ask if I should retrain the VAE module for my work and how to train the VAE module.
